app/Actions/Server/DeleteServer.php

```
@@ -2,16 +2,102 @@
 use App\Models\Server;
+use App\Models\Team;
+use App\Notifications\Server\HetznerDeletionFailed;
+use App\Services\HetznerService;
 use Lorisleiva\Actions\Concerns\AsAction;
 
 class DeleteServer
 {
     use AsAction;
 
-    public function handle(Server $server)
+    public function handle(int $serverId, bool $deleteFromHetzner = false, ?int $hetznerServerId = null, ?int $cloudProviderTokenId = null, ?int $teamId = null)
```

PHPMD (2.15.0)

The method handle has a boolean flag argument $deleteFromHetzner, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)